lambdas/cli.py

```diff
@@ -0,0 +1,283 @@
+        ctx.exit(1)
+
+
+@cli.command()
+@click.pass_context
+@click.option("--aip-uuid", "-a", required=False, help="AIP UUID from Archivematica.")
+@click.option("--s3-uri", "-u", required=False, help="Full S3 URI of AIP stored in S3.")
+@click.option(
+    "--details",
+    "-d",
+    required=False,
+    is_flag=True,
+    help="Return full AIP validation details as JSON to stdout instead of 'OK'.",
+)
+def validate(ctx: click.Context, aip_uuid: str, s3_uri: str, *, details: bool) -> None:
+    """Validate a single AIP stored in S3 via the AIP UUID or S3 URI.
+
+    The result is either 'OK' or the full validation response if the '--details' is set.
+
+    Note: the timeout for the lambda HTTP request is quite long to accommodate AIPs that
+    take substantial time to validate.  If there are connection issues it is recommended
+    to use the 'ping' CLI command to ensure firewall access and authorization.
+    """
+    if not (aip_uuid or s3_uri):
+        raise click.UsageError("You must provide either --aip-uuid/-a or --s3-uri/-u.")
+
+    logger.debug("Starting AIP validation")
+
+    # make request to AIP validator lambda
+    try:
+        result = validate_aip_via_lambda(
+            aip_uuid=aip_uuid, aip_s3_uri=s3_uri, verbose=ctx.obj["VERBOSE"]
+        )
+    except requests.exceptions.RequestException as exc:
+        error_msg = f"Error connecting to AIP validation lambda: {exc}"
+        logger.error(error_msg)
+        click.echo(error_msg, err=True)
+        ctx.exit(1)
+    except ValueError as exc:
+        error_msg = f"Error with validation parameters: {exc}"
+        logger.error(error_msg)
+        click.echo(error_msg, err=True)
+        ctx.exit(1)
+
+    if elapsed := result.get("elapsed"):
+        logger.debug(f"AIP validation elapsed: {elapsed}s")
+
+    # send results to stdout and set exit codes
+    if result.get("valid"):
+        if details:
+            click.echo(json.dumps(result))
+        else:
+            click.echo("OK")
+        ctx.exit(0)
+    else:
+        click.echo(f"AIP validation error: {result.get('error', 'Unknown')}", err=True)
+        ctx.exit(1)
+
+
+@cli.command()
+@click.pass_context
+@click.option(
+    "--input-csv-filepath",
+    "-i",
+    required=True,
+    help="Filepath of CSV with AIP UUIDs or S3 URIs to be validated.",
+)
+@click.option(
+    "--output-csv-filepath",
+    "-o",
+    required=False,
+    help="Filepath of CSV for validation results.",
+)
+@click.option(
+    "--details",
+    "-d",
+    required=False,
+    is_flag=True,
+    help="Return full AIP validation details as JSON to stdout instead of 'OK'.",
+)
+@click.option(
+    "--max-workers",
+    "-w",
+    required=False,
+    type=int,
+    default=25,
+    envvar="LAMBDA_MAX_CONCURRENCY",
+    help=(
+        "Maximum number of concurrent validation workers.  This should not exceed the "
+        "maximum concurrency for the deployed AWS Lambda function."
+    ),
+)
+def bulk_validate(
```

Comment:
Do you think it would also be helpful to provide a Click option where `multiple=True` to provide AIP UUIDs or S3 URIs (i.e., not require the creation of an input CSV file)? 🤔 

```text
bulk-validate -a AAA -a BBB -a CCC ...
```

Reply:
I had considered it, but I think it's either going to be **one** AIP to validate or like **one hundred**.

I should have communicated this more clearly in the PR, and maybe even in code, but the CLI is really kind of a stopgap; it should _not_ be the primary way in which this tool is used longterm.  

And even if it _is_, I think programatically calling it from another script would be most practical.